external/sql/storm-sql-core/src/test/org/apache/storm/sql/compiler/TestCompilerUtils.java

```diff
@@ -153,6 +158,29 @@ public static CalciteState sqlOverNestedTable(String sql)
                 .field("MAPFIELD", SqlTypeName.ANY)
                 .field("NESTEDMAPFIELD", SqlTypeName.ANY)
                 .field("ARRAYFIELD", SqlTypeName.ANY)
+
+                // TODO: replace field definitions with belows while Calcite can handle properly
```

Comment:
For me, type safely is more important than convenience. That's why we use static typing language, and also trying to restrict more by generic and so. IMO, we shouldn't rely Calcite to handle operators with 'ANY' type properly. If it can handle operators properly, it's more than essential feature.

As I commented earlier we can't keep the current syntax until we modify Calcite, or back to ExprCompiler which I really don't want to.
For now I'll try to CAST to 'ANY' explicitly and see it makes nested multi-level access possible.